plugins/ssl/certbot_certs

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/perl 
+# -*- perl -*-
+
+=encoding utf8
+
+=head1 NAME
+
+certbot_certs - Munin plugin to certbot (letsencrypt) certificate expiry
+
+=head1 APPLICABLE SYSTEMS
+
+Servers running certbot certificates.  This script auto-discovers the current set of certificates
+
+=head1 CONFIGURATION
+
+This script requires root permissions to run, in your munin-node configuration:
+
+  [certbot_certs]
+  user root
+
+=head1 CAPABILITIES
+
+This plugin supports L<DIRTYCONFIG|http://guide.munin-monitoring.org/en/latest/plugin/protocol-dirtyconfig.html>
+
+=head1 LICENSE
+
+Copyright (C) 2018 J.T.Sage (jtsage@gmail.com)
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see L<http://www.gnu.org/licenses/>.
+
+=head1 MAGIC MARKERS
+
+  #%# family=manual
+  #%# capabilities=
+
+=cut
+
+use warnings;
+use strict;
+use Munin::Plugin;
+use Date::Parse;
+use POSIX;
+
+if ( defined($ARGV[0]) && $ARGV[0] eq "autoconf" ) {
+	if ( -e "/usr/bin/openssl" ) {
+		print "no (Root User Required)\n";
```

Comment:
You do not accept *yes* for an answer? :)

Reply:
Nope, but it's possible I totally misunderstood autoconf - I was under the impression that if it returns yes, it can be added to the munin-node configuration without any user intervention? (In this case, it requires elevated permissions to read the certificates).  If i'm wrong, happy to make that edit.

Re: name change - yes, good call.

Reply:
It is possible, that the plugin is pre-configured somehow (e.g. the Debian package for munin contains the necessary plugin configuration for all distributed plugins).

Usually plugins check whether dependencies are missing and if runtime conditions are met. In your case you could check whether the directory `/etc/letsencrypt/renewal` exists and is readable (both should be reported separately, if missing).

Reply:
That makes sense.  I fleshed out that logic a bit, and added a number of other traps to keep the plugin sane.  Thanks for the input.